006-admin-server.md

```diff
@@ -0,0 +1,116 @@
+# Strimzi Admin Server
+
+The Strimzi Admin repository was setup in December 2019 to hold an implementation of an Admin API. The repository has not had any contributions yet so this proposal sets out what the structure of an Admin Server might look like in order to get to the initial implementation of the server.
+
+## Motivation
+
+An API server will:
+* provide a consolidated backend API for additional interfaces like a browser based UI or a CLI.
+* be capable of supporting both REST and GraphQL interfaces.
+* allow for more sophisticated APIs to be built on top of the existing APIs like the kubernetes API providing value-add to existing content.
+
+## Proposal
+
+### The server
+The proposal is to make a modular server using the Vert.x-Web toolkit which has strong support for creating both REST and GraphQL interfaces.
```

Comment:
What is the value and the cost of supporting both REST and GraphQL? Not just for development, but also QE, maintenance etc.

Reply:
I do not think it will add a huge amount of overhead supporting both. Both APIs will be retrieving similar data but in a slightly different way. This allows the collection and aggregation of the information to be done in a way that is common to both and the difference then becomes a small piece on top that formats the results slightly differently. Both apis still use a REST endpoint with the main difference being that the REST api has multiple resource paths and each path represents a resource whereas GraphQL has a single resource path and the different resources to be returned are defined in the payload so similar tools can be used to test both APIs. Both APIs will also be implemented using the Vert.x libraries so keeping the libraries in-sync should be straightforward.